there can be only vmem!

kilobyte opened the pull request:
😼

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/vmem/1)
<!-- Reviewable:end -->

marcinslusarz committed:
Just VMEM.

kilobyte committed:
test: ignore tests not named ^TEST[0-9]+$

Fixes breakage on files named TEST17~ or TESTS.py

The RUNTESTS script already requires the name to be a number, so there's
no loss of generality.

kilobyte committed:
Disable packaging on Travis.

kilobyte committed:
common: rename the main MSVC "solution" to VMEM

kilobyte committed:
test: use system-installed pmempool or skip

The VMEM project doesn't have pmempool anymore; you may or may not use PMDK
as a build-dependency.

kilobyte committed:
Update MSVC "solution" files for vmem extraction.

kilobyte committed:
Drop the separate examples build (already done via VMEM.sln).

kilobyte committed:
Don't put non-vmem stuff into zips.

kilobyte committed:
Don't try to run pmemdetect.

kilobyte committed:
test: adjust matches for out_err_mt_win

kilobyte committed:
test: drop util_uuid_generate

kilobyte committed:
test: initialize libpmemcommon

This was done in the past by libpmem's constructor which libut linked to.

kilobyte committed:
test: drop win_poolset_unmap

kilobyte committed:
test: drop win_getopt, win_mmap, win_mmap_fixed

Got accidentally readded during rebases.

kilobyte committed:
test: don't fail a devdax test; always skip it for now

We'd need to re-implement devdax detection some way.

kilobyte committed:
Fix a misbuild in MSCV "solution".

kilobyte committed:
test: license dates

kilobyte committed:
common: drop pmempool stuff introduced after 1.5

kilobyte committed:
test: drop pmem_deep_persist tests

kilobyte committed:
Drop ndctl and libfabric stuff.

kilobyte committed:
common: unblock unknown architectures

There's no arch-specific bits in vmem.  It most likely requires 64-bit, and
possibly little-endian, but that's it.

kilobyte committed:
common: drop unused arch_flags logic

kilobyte committed:
doc: remove PMDK-specific bits from README

kilobyte committed:
Trim distro builds.

kilobyte committed:
Clean away ndctl from docker builds.

kilobyte committed:
test: vmem: update valgrind matches for glibc-2.29

kilobyte committed:
Drop CONTRIBUTING.

A good part doesn't fit vmem, and none of our other projects have this doc,
thus coming up with new material is not worth the effort.

kilobyte committed:
Documentation updates.

kilobyte committed:
Drop the area prefix commit subject requirement.

Too few areas left...

kilobyte committed:
Adjust MSCV "solution" for 1.5.2..1.7 merge.

kilobyte committed:
Merge in 1.7-rc2

kilobyte committed:
Drop remnants of the Python framework -- not used for vmem.

kilobyte committed:
Set FAULT_INJECTION on libut.

In PMDK libut linked against libpmem which pulled in the _flt symbols.

kilobyte committed:
Don't run old ndctl tests.

vmem doesn't care about ndctl.

kilobyte committed:
Purge ndctl/libfabric stuff.

kilobyte committed:
Fix distro builds.

kilobyte committed:
Drop non-existent man page sections.

The build system doesn't tolerate them missing.

kilobyte committed:
Prune stuff introduced by the 1.7 merge.

kilobyte committed:
Merge with the final 1.7 release of PMDK.

marcinslusarz commented on README.md:
```diff
@@ -1,95 +1,59 @@
 ## Pre-Built Packages for Windows
 
-The recommended and easiest way to install PMDK on Windows is to use Microsoft vcpkg. Vcpkg is an open source tool and ecosystem created for library management.
+The recommended and easiest way to install VMEM on Windows is to use Microsoft vcpkg. Vcpkg is an open source tool and ecosystem created for library management.
```
There's no VMEM in vcpkg yet.

kilobyte replied:
I've marked that "yet".

marcinslusarz commented on README.md:
```diff
@@ -283,7 +234,7 @@ run different types of tests.
 ### Memory Management Tools
 
-The PMDK libraries support standard Valgrind DRD, Helgrind and Memcheck, as well as a PM-aware version of [Valgrind](https://github.com/pmem/valgrind) (not yet available for FreeBSD). By default, support for all tools is enabled. If you wish to disable it, supply the compiler with **VG_\<TOOL\>_ENABLED** flag set to 0, for example:
+The VMDK libraries support standard Valgrind DRD, Helgrind and Memcheck, as well as a PM-aware version of [Valgrind](https://github.com/pmem/valgrind) (not yet available for FreeBSD). By default, support for all tools is enabled. If you wish to disable it, supply the compiler with **VG_\<TOOL\>_ENABLED** flag set to 0, for example:
```
VMDK?

kilobyte replied:
✓

marcinslusarz commented:
- VERSION shouldn't exist
- the merge of src/Makefile looks suspicious - ALL_TARGETS contains examples, where previously it didn't, RELEASE_TARGETS - the same
- src/Makefile - pkg-cfg-common generates rasdeps variable
- src/common/.cstyleignore - what's up with pmemcompat.h change?
- src/test/Makefile - envconfig changes seem to be wrong
- src/test/Makefile.inc - uuid.o is commented out, should be removed?, vpath is commented out
- src/test/RUNTESTS - pmemdetect and pmem_fs_dir support is commented out
- src/test/vmem_malloc/TEST2 - that's cheating
- utils/build-dpkg.sh - $PACKAGE_NAME should probably be changed

We need separate Docker images for this repo.

kilobyte committed:
More naming changes.

kilobyte committed:
No Vcpkg yet.

kilobyte committed:
Don't build examples when requested not to.

kilobyte committed:
Some more purging.

kilobyte committed:
Delete VERSION.

kilobyte commented:
VERSION: ✓.
ALL_TARGETS: ✓.
rasdeps: ✓.
.cstyleignore for pmemcompat.h: same in PMDK, cstyle can't handle templates in this file.
envconfig: please elaborate.
uuid.o, vpath: ✓.
pmemdetect: let's discuss: what's the point of separate pmem/nonpmem tests?
cheating: will need to find devdax some way.
PACKAGE_NAME: only place that sets it, says "vmem".  That's ok?
Docker: let's trick someone else into contributing ☺

marcinslusarz commented:
src/Makefile - what about RELEASE_TARGETS?
envconfig - %s without argument
src/test/Makefile.inc - $(TOP)/src/nondebug/common/util_posix.o**_**
pmemdetect - I'm not saying "we should have such tests", I'm saying "don't leave commented out code"
cheating - ok, I'm waiting
PACKAGE_NAME - yeah, I missed this change
Docker - it's trivial, but I can do this after the merge
src/common/common.rc - VMDK again?

kilobyte committed:
Drop a stray underscore.

kilobyte committed:
Eradicate envconfig.

Was used only for ndctl libs.

kilobyte committed:
Drop no longer used RELEASE_TARGETS.

kilobyte commented:
RELEASE_TARGETS: unused, removed.
envconfig: was used only for ndctl libs, removed.
util_.posix.o**_**: fixed
VMDK: not in my tree, which was in sync with Github.

marcinslusarz commented:
```
-                       VALUE "ProductName", "Persistent Memory Development Kit"
+                       VALUE "ProductName", "Volatile Memory Development Kit"
```

kilobyte committed:
Restore vmem_malloc/TEST2 (devdax).

kilobyte committed:
A better ProductName.
